syntax = "proto2";

package SimpleChat;

enum NetworkMessageType {
    USER_JOIN_REQUEST = 0;
    USER_JOIN_RESPONSE = 1;
    USER_LIST_REQUEST = 2;
    USER_LIST_RESPONSE = 3;
    USER_CHANGE = 4;
    
    CHAT_MESSAGE = 10;
    CHAT_AUTHORIZE = 11;
    CHAT_COMMAND = 12;

    GENERIC_CHAT_RESPONSE = 20;

    CHATROOM_CHANGE = 25;
}

message NetworkMessageHeader {
    required NetworkMessageType type = 1;
}

message NetworkMessage {
    required NetworkMessageHeader header = 1;
    required bytes serialized_data = 2;

    extensions 10 to 200;
}